src/vmm/src/resources.rs

```diff
@@ -128,6 +124,10 @@ pub struct VmResources {
     pub mmds_size_limit: usize,
     /// Whether or not to load boot timer device.
     pub boot_timer: bool,
+    /// Template to use for configuring vCPUs.
+    /// Will not be used in conjunction with a hard-coded
+    /// template as defined by CpuFeaturesTemplate
+    pub guest_cpu_template: Option<CustomCpuTemplateType>,
```

Comment:
> guest it is ok to just name this cpu_template.

I don't have a strong opinion on the attribute name. I did make it `guest_*` here because the `VmResources` struct contains VcpuConfig which has the static/hard-coded template definition has well, and wanted some way to differentiate between the types. What would your preference be, is this reason sufficient for the `guest_` name style?

> Also as per the previous commit, CustomCpuTemplateType is the actual template not the type so why not just say Option<CustomCpuTemplate>?

I wanted to make it clear that the `*Type` field is the only alias, and that is not specific about which CPU architecture is used. I don't think it helps in this particular case, I will change the alias name to `CustomCpuTemplate`.